src/RemoteResourceController.js

```diff
@@ -16,16 +16,59 @@
 
 const request = require('request-promise-native');
 const clone = require('clone');
+const objectPath = require('object-path');
 
 const { BaseDownloadController } = require('@razee/razeedeploy-core');
 
-
 module.exports = class RemoteResourceController extends BaseDownloadController {
   constructor(params) {
     params.finalizerString = params.finalizerString || 'children.remoteresource.deploy.razee.io';
     super(params);
   }
 
+  async added() {
+    let requests = objectPath.get(this.data, ['object', 'spec', 'requests'], []);
+    let newRequests = [];
+    for (let i = 0; i < requests.length; i++) {
+      let r = requests[i];
+      let headers = objectPath.get(r, 'options.headers');
+      if (headers) {
+        let headerKeysList = Object.keys(headers);
+        for (let j = 0; j < headerKeysList.length; j++) {
+          let headerName = headerKeysList[j];
+          let headerObject = headers[headerName];
```

Comment:
try this instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries
```suggestion
        for (let [headerName, headerObject] of Object.entries(headers)) {
```